src/govuk/components/accordion/accordion.test.js

```diff
@@ -230,32 +231,57 @@ describe('/components/accordion', () => {
         })
       })
 
-      it('should change the (combined) Show[ this section] text to Hide[ this section ] when sections are opened', async () => {
+      it('should change the Show text to Hide when sections are opened', async () => {
         await goToComponent(page, 'accordion')
         await page.click('.govuk-accordion .govuk-accordion__section:nth-of-type(2) .govuk-accordion__section-header')
         await page.click('.govuk-accordion .govuk-accordion__section:nth-of-type(3) .govuk-accordion__section-header')
 
         const ShowOrHideButtonText = await page.evaluate(() => document.body.querySelector('.govuk-accordion__section-toggle-text').textContent)
 
-        expect(ShowOrHideButtonText).toEqual('Hide this section')
+        expect(ShowOrHideButtonText).toEqual('Hide')
       })
 
-      it('should have a data-nosnippet attribute on the "Show / hide this section" container to hide it from search result snippets', async () => {
+      it('should have a data-nosnippet attribute on the "Show / hide" container to hide it from search result snippets', async () => {
         await goToComponent(page, 'accordion')
 
         const dataNoSnippetAttribute = await page.evaluate(() => document.body.querySelector('.govuk-accordion__section-toggle').getAttribute('data-nosnippet'))
 
         expect(dataNoSnippetAttribute).toEqual('')
       })
 
-      describe('hidden text', () => {
-        it('should contain hidden text " this section" for assistive technology', async () => {
+      describe('accessible name', () => {
+        it('Should set the accessible name of the show/hide section buttons', async () => {
           await goToComponent(page, 'accordion')
 
-          const numberOfExampleSections = 2
-          const hiddenText = await page.evaluate(() => document.body.querySelectorAll('.govuk-accordion .govuk-accordion__section .govuk-accordion__section-toggle-text .govuk-visually-hidden').length)
+          const element = await page.$('.govuk-accordion__section-button')
+
+          expect(getAccessibleName(element)).resolves.toBe(
+            'Section A , Show this section'
+          )
+
+          await element.click()
+
+          expect(getAccessibleName(element)).resolves.toBe(
+            'Section A , Hide this section'
+          )
+        })
+
+        it('Includes the heading summary', async () => {
+          await goToComponent(page, 'accordion', {
+            exampleName: 'with-additional-descriptions'
+          })
+
+          const element = await page.$('.govuk-accordion__section-button')
+
+          expect(getAccessibleName(element)).resolves.toBe(
+            'Test , Additional description , Show this section'
+          )
 
-          expect(hiddenText).toEqual(numberOfExampleSections)
+          await element.click()
+
+          expect(getAccessibleName(element)).resolves.toBe(
```

Comment:
This is superb 🎉 

Proves that the SAME element has newly updated accessible text